src/apps/components/juttle-viewer.js

```diff
@@ -1,9 +1,43 @@
 import React from 'react';
 
-const JuttleViewer = (props) => {
-    return (
-        <pre className='juttle-source'>{props.bundle.program}</pre>
-    );
+import JuttleEditor from './juttle-editor';
+import { newBundle } from '../actions';
+import { LOCAL_BUNDLE_ID } from '../constants';
+import { Link } from 'react-router';
+
+class JuttleViewer extends React.Component {
+    constructor(props) {
+        super(props);
+        let isNewBundle = !props.bundle;
+        if (isNewBundle) {
+            props.dispatch(newBundle(LOCAL_BUNDLE_ID, {program: ''}, {}));
+        }
+    }
```

Comment:
I'm not a fan of this. 
- If there is no bundle, the JuttleViewer should never render and the code button should be disable. 
- I worry this would put the application in a weird state where we're in path mode but the bundleId is set to `local_bundle_id` which should really only be the case if the runMode is local
- In general views should only be changing redux state as a result of actions from the user. This would make it difficult for a developer to find out where the action is getting called.